src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt

```diff
@@ -60,9 +62,9 @@ class WalletController(
                         onboardingChannel = xClientIdDto.toOnboardingChannel()
                     )
                     .flatMap { (loggedAction, returnUri) ->
-                        loggedAction.saveEvents(loggingEventRepository).map {
-                            Triple(it.id.value, request, returnUri)
-                        }
+                        Mono.defer { walletEventSinksService.tryEmitEvent(loggedAction) }
+                            .map { Triple(it.data.id.value, request, returnUri) }
+                            .onErrorReturn(Triple(loggedAction.data.id.value, request, returnUri))
```

Comment:
this comment applies also to the other controller methods below too

Reply:
Right, it can be simplified but I prefer adding error handling to the caller too to make it not dependent on the implementation of `tryEmitEvent`; this method could change the way it handles errors in the future without breaking the caller. What do you think about this approach?

Reply:
Since the entire code base is in the same project I would remove this duplicated condition just to make code more readable.
Moreover there are junit tests that would explode in case of code modification so an unintentional modification should never happen here

Reply:
Done in [2c54e5e](https://github.com/pagopa/pagopa-wallet-service/pull/195/commits/2c54e5e678106b13117270fbfa85984475000aee)